File: restaurant-app-client/src/app/models/orderDto.ts
import { OrderService } from '../services/order/order.service';
import { Menu } from './menuDto';

export class Order{
    "id" : number;
    "orderType" : string;
    "orderStatus" : string;
    "price" : number;
    "customerId" : number;
    "tableId" : number;
    "menuItems" : number[]
    "menuItemsDto" : Menu[]
}

File: restaurant-app-server/src/main/java/com/example/restaurantappserver/api/dto/OrderDto.java
package com.example.restaurantappserver.api.dto;

import com.example.restaurantappserver.entity.MenuItem;
import com.example.restaurantappserver.enums.OrderStatus;
import com.example.restaurantappserver.enums.OrderType;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class OrderDto {

    private Long id;
    private OrderType orderType;
    private OrderStatus orderStatus;

    private Double price;
    private Long customerId;
    private UserDto customer;


    private Long tableId;
    private TableDto tableDto;

    private List<Long> menuItems;
    private List<MenuItemDto> menuItemsDto;

}
